Unable to build the driver

Hi, I'm fairly new to GitHub and downloading codes and compiling them for use, so this may be something I'm not doing right.

I'm unable to build the driver using the method described "gradlew clean build" getting errors at different stages. The last try listed out about 16 failures during the build process. The first error message was at: 

AsyncStreamingResultSetTest > testThrowsOnNonsenseQueryAsync() FAILED
    org.opentest4j.AssertionFailedError at AsyncStreamingResultSetTest.java:45
        Caused by: org.opentest4j.AssertionFailedError at AsyncStreamingResultSetTest.java:46

Attaching the error and problem reports. 

[problems-report.txt](https://github.com/user-attachments/files/20151345/problems-report.txt)

[test.zip](https://github.com/user-attachments/files/20151357/test.zip)

--- a/.github/workflows/reusable-build-publish.yml
+++ b/.github/workflows/reusable-build-publish.yml
@@ -44,7 +44,7 @@ jobs:
     needs: prepare
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest]
+        os: [ ubuntu-latest, windows-latest ]
       fail-fast: false
     runs-on: ${{ matrix.os }}
     steps:
@@ -76,59 +76,54 @@ jobs:
       - name: Extract Hyper
         id: extract_hyper
         run: ./gradlew extractHyper
-      - name: Look at files in root
+      - name: Gradle build (with version)
+        id: build-and-test
+        run: |
+          echo "build and test with version='${{ inputs.version }}'"
+          ./gradlew clean build --no-build-cache --rerun-tasks
+        env:
+          RELEASE_VERSION: ${{ inputs.version }}
+      - name: Look at files in verification
+        if: ${{ failure() }}
         uses: jaywcjlove/github-action-folder-tree@main
         with:
           path: .
-          depth: 2
-      - name: Intentionally fail
-        run: exit 1
-      # - name: Gradle build (with version)
-      #   id: build-and-test
-      #   run: |
-      #     echo "build and test with version='${{ inputs.version }}'"
-      #     ./gradlew clean build --no-build-cache --rerun-tasks
-      #   env:
-      #     RELEASE_VERSION: ${{ inputs.version }}
-      # - name: Look at files in verification
-      #   if: ${{ failure() }}
-      #   uses: jaywcjlove/github-action-folder-tree@main
-      #   with:
-      #     path: verification
-      # - name: Upload hyper logs on failure
-      #   if: ${{ failure() }}
-      #   uses: actions/upload-artifact@v4
-      #   with:
-      #     name: test-results-${{ matrix.os }}
-      #     path: |
-      #       build/hyperd/*.log
-      #     retention-days: 5
+          exclude: ".gradle|build|bin"
+          depth: 3
+      - name: Upload hyper logs on failure
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-results-${{ matrix.os }}
+          path: |
+            build/hyperd/*.log
+          retention-days: 5
 
-  # publish:
-  #   needs: [prepare, build]
-  #   if: ${{ inputs.publish }}
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v4
-  #       with:
-  #         fetch-depth: 1
-  #     - run: git fetch --no-tags --depth=1 origin main
-  #     - name: Setup Java with 8 and 17 toolchains
-  #       uses: actions/setup-java@v4
-  #       with:
-  #         distribution: 'temurin'
-  #         java-version: |
-  #           8
-  #           17
-  #     - name: Setup Gradle
-  #       uses: gradle/actions/setup-gradle@v4
-  #       with:
-  #         gradle-version: ${{ env.GRADLE_VERSION }}
-  #     - name: Publish to Maven Central
-  #       env:
-  #         RELEASE_VERSION: ${{ inputs.version }}
-  #         OSSRH_USERNAME: ${{ secrets.publish_user }}
-  #         OSSRH_PASSWORD: ${{ secrets.publish_pass }}
-  #         ORG_GRADLE_PROJECT_signingKey: ${{ secrets.signing_key }}
-  #         ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.signing_pass }}
-  #       run: ./gradlew publishAllPublicationsToMavenCentralRepository -x test --no-scan --no-configuration-cache -Dorg.slf4j.simpleLogger.defaultLogLevel=error
+  publish:
+    needs: [ prepare, build ]
+    if: ${{ inputs.publish }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - run: git fetch --no-tags --depth=1 origin main
+      - name: Setup Java with 8 and 17 toolchains
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: |
+            8
+            17
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          gradle-version: ${{ env.GRADLE_VERSION }}
+      - name: Publish to Maven Central
+        env:
+          RELEASE_VERSION: ${{ inputs.version }}
+          OSSRH_USERNAME: ${{ secrets.publish_user }}
+          OSSRH_PASSWORD: ${{ secrets.publish_pass }}
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.signing_key }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.signing_pass }}
+        run: ./gradlew publishAllPublicationsToMavenCentralRepository -x test --no-scan --no-configuration-cache -Dorg.slf4j.simpleLogger.defaultLogLevel=error

--- a/.github/workflows/reusable-build-publish.yml
+++ b/.github/workflows/reusable-build-publish.yml
@@ -88,7 +88,7 @@ jobs:
         uses: jaywcjlove/github-action-folder-tree@main
         with:
           path: .
-          exclude: ".gradle|build|bin"
+          exclude: ".gradle|build|bin|.git"
           depth: 3
       - name: Upload hyper logs on failure
         if: ${{ failure() }}

--- a/.github/workflows/reusable-build-publish.yml
+++ b/.github/workflows/reusable-build-publish.yml
@@ -62,7 +62,7 @@ jobs:
       - name: Cache hyperd zip
         uses: actions/cache@v3
         with:
-          path: build/hyper-${{ needs.prepare.outputs.hyper_version }}.zip
+          path: .hyper/hyper-${{ needs.prepare.outputs.hyper_version }}.zip
           key: ${{ runner.os }}-hyper-${{ needs.prepare.outputs.hyper_version }}
           restore-keys: ${{ runner.os }}-hyper-${{ needs.prepare.outputs.hyper_version }}
       - name: Setup Gradle

--- a/.github/workflows/reusable-build-publish.yml
+++ b/.github/workflows/reusable-build-publish.yml
@@ -73,55 +73,62 @@ jobs:
           build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
           build-scan-terms-of-use-agree: "yes"
           gradle-version: ${{ env.GRADLE_VERSION }}
-      - name: Gradle build (with version)
-        id: build-and-test
-        run: |
-          echo "build and test with version='${{ inputs.version }}'"
-          ./gradlew clean build --no-build-cache --rerun-tasks
-        env:
-          RELEASE_VERSION: ${{ inputs.version }}
-      - name: Look at files in verification
-        if: ${{ failure() }}
-        run: |
-          if [ "$RUNNER_OS" == "Windows" ]; then
-            gci -Recurse verification || true
-          else
-            ls -la verification || true
-          fi
-      - name: Upload hyper logs on failure
-        if: ${{ failure() }}
-        uses: actions/upload-artifact@v4
+      - name: Extract Hyper
+        id: extract_hyper
+        run: ./gradlew extractHyper
+      - name: Look at files in root
+        uses: jaywcjlove/github-action-folder-tree@main
         with:
-          name: test-results-${{ matrix.os }}
-          path: |
-            build/hyperd/*.log
-          retention-days: 5
+          path: .
+          depth: 2
+      - name: Intentionally fail
+        run: exit 1
+      # - name: Gradle build (with version)
+      #   id: build-and-test
+      #   run: |
+      #     echo "build and test with version='${{ inputs.version }}'"
+      #     ./gradlew clean build --no-build-cache --rerun-tasks
+      #   env:
+      #     RELEASE_VERSION: ${{ inputs.version }}
+      # - name: Look at files in verification
+      #   if: ${{ failure() }}
+      #   uses: jaywcjlove/github-action-folder-tree@main
+      #   with:
+      #     path: verification
+      # - name: Upload hyper logs on failure
+      #   if: ${{ failure() }}
+      #   uses: actions/upload-artifact@v4
+      #   with:
+      #     name: test-results-${{ matrix.os }}
+      #     path: |
+      #       build/hyperd/*.log
+      #     retention-days: 5
 
-  publish:
-    needs: [prepare, build]
-    if: ${{ inputs.publish }}
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 1
-      - run: git fetch --no-tags --depth=1 origin main
-      - name: Setup Java with 8 and 17 toolchains
-        uses: actions/setup-java@v4
-        with:
-          distribution: 'temurin'
-          java-version: |
-            8
-            17
-      - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@v4
-        with:
-          gradle-version: ${{ env.GRADLE_VERSION }}
-      - name: Publish to Maven Central
-        env:
-          RELEASE_VERSION: ${{ inputs.version }}
-          OSSRH_USERNAME: ${{ secrets.publish_user }}
-          OSSRH_PASSWORD: ${{ secrets.publish_pass }}
-          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.signing_key }}
-          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.signing_pass }}
-        run: ./gradlew publishAllPublicationsToMavenCentralRepository -x test --no-scan --no-configuration-cache -Dorg.slf4j.simpleLogger.defaultLogLevel=error
+  # publish:
+  #   needs: [prepare, build]
+  #   if: ${{ inputs.publish }}
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #       with:
+  #         fetch-depth: 1
+  #     - run: git fetch --no-tags --depth=1 origin main
+  #     - name: Setup Java with 8 and 17 toolchains
+  #       uses: actions/setup-java@v4
+  #       with:
+  #         distribution: 'temurin'
+  #         java-version: |
+  #           8
+  #           17
+  #     - name: Setup Gradle
+  #       uses: gradle/actions/setup-gradle@v4
+  #       with:
+  #         gradle-version: ${{ env.GRADLE_VERSION }}
+  #     - name: Publish to Maven Central
+  #       env:
+  #         RELEASE_VERSION: ${{ inputs.version }}
+  #         OSSRH_USERNAME: ${{ secrets.publish_user }}
+  #         OSSRH_PASSWORD: ${{ secrets.publish_pass }}
+  #         ORG_GRADLE_PROJECT_signingKey: ${{ secrets.signing_key }}
+  #         ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.signing_pass }}
+  #       run: ./gradlew publishAllPublicationsToMavenCentralRepository -x test --no-scan --no-configuration-cache -Dorg.slf4j.simpleLogger.defaultLogLevel=error

--- a/.github/workflows/reusable-build-publish.yml
+++ b/.github/workflows/reusable-build-publish.yml
@@ -22,8 +22,80 @@ on:
       publish_pass:
         required: false
 
+env:
+  GRADLE_VERSION: "8.12"
+
 jobs:
-  build-and-publish:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      hyper_version: ${{ steps.get_version.outputs.hyper_version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: Get hyperd version
+        id: get_version
+        run: |
+          HYPER_VERSION=$(sed -n 's/^hyperApiVersion=\(.*\)/\1/p' gradle.properties)
+          echo "hyper_version=$HYPER_VERSION" >> $GITHUB_OUTPUT
+
+  build:
+    needs: prepare
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - run: git fetch --no-tags --depth=1 origin main
+      - name: Setup Java with 8 and 17 toolchains
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: |
+            8
+            17
+      - name: Cache hyperd zip
+        uses: actions/cache@v3
+        with:
+          path: build/hyper-${{ needs.prepare.outputs.hyper_version }}.zip
+          key: ${{ runner.os }}-hyper-${{ needs.prepare.outputs.hyper_version }}
+          restore-keys: ${{ runner.os }}-hyper-${{ needs.prepare.outputs.hyper_version }}
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          dependency-graph: generate-and-submit
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
+          gradle-version: ${{ env.GRADLE_VERSION }}
+      - name: Gradle build (with version)
+        id: build-and-test
+        run: |
+          echo "Version specified, running clean build with no cache for ${{ inputs.version }}"
+          ./gradlew clean build --no-build-cache --rerun-tasks
+        env:
+          RELEASE_VERSION: ${{ inputs.version }}
+      - name: Look at files in verification
+        if: ${{ failure() }}
+        run: |
+          ls -la verification || true
+      - name: Upload hyper logs on failure
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-results-${{ matrix.os }}
+          path: |
+            build/hyperd/*.log
+          retention-days: 5
+
+  publish:
+    needs: [prepare, build]
+    if: ${{ inputs.publish }}
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
@@ -37,57 +109,15 @@ jobs:
           java-version: |
             8
             17
-      - name: Get hyperd version
-        run: |
-          HYPER_VERSION=$(sed -n 's/^hyperApiVersion=\(.*\)/\1/p' gradle.properties)
-          echo "HYPER_VERSION=$HYPER_VERSION" >> $GITHUB_ENV
-      - name: Cache hyperd zip
-        uses: actions/cache@v3
-        with:
-          path: build/hyper-${{ env.HYPER_VERSION }}.zip
-          key: ${{ runner.os }}-hyper-${{ env.HYPER_VERSION }}
-          restore-keys: ${{ runner.os }}-hyper-${{ env.HYPER_VERSION }}
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v4
         with:
-          dependency-graph: generate-and-submit
-          build-scan-publish: true
-          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
-          build-scan-terms-of-use-agree: "yes"
-          gradle-version: "8.12"
-      - name: Gradle build
-        id: build
-        run: |
-          if [[ "${{ inputs.version }}" != "" ]]; then
-            echo "Version specified, running clean build with no cache for ${{ inputs.version }}"
-            gradle clean check --no-build-cache --rerun-tasks
-          else
-            echo "No version specified, running normal build"
-            gradle clean check -Dorg.slf4j.simpleLogger.defaultLogLevel=error
-          fi
-        env:
-          RELEASE_VERSION: ${{ inputs.version }}
-      - name: Publish on release
-        id: publish
-        if: ${{ inputs.publish && (success() || steps.build.outcome == 'success') }}
+          gradle-version: ${{ env.GRADLE_VERSION }}
+      - name: Publish to Maven Central
         env:
           RELEASE_VERSION: ${{ inputs.version }}
           OSSRH_USERNAME: ${{ secrets.publish_user }}
           OSSRH_PASSWORD: ${{ secrets.publish_pass }}
-          # https://docs.gradle.org/current/userguide/signing_plugin.html#sec:in-memory-keys
-          # https://central.sonatype.org/publish/requirements/gpg/#generating-a-key-pair
-          # gpg --export-secret-key --armor <key id> | pbcopy
           ORG_GRADLE_PROJECT_signingKey: ${{ secrets.signing_key }}
           ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.signing_pass }}
-        run: gradle publishAllPublicationsToMavenCentralRepository --no-scan --no-configuration-cache -Dorg.slf4j.simpleLogger.defaultLogLevel=error
-      - name: Look at files in verification
-        if: ${{ failure() || steps.build.outcome == 'failure' || steps.publish.outcome == 'failure' }}
-        run: ls -lR verification
-      - name: Upload hyper logs on failure
-        if: ${{ failure() || steps.build.outcome == 'failure' || steps.publish.outcome == 'failure' }}
-        uses: actions/upload-artifact@v4
-        with:
-          name: test-results
-          path: |
-            build/hyperd/*.log
-          retention-days: 5
+        run: ./gradlew publishAllPublicationsToMavenCentralRepository -x test --no-scan --no-configuration-cache -Dorg.slf4j.simpleLogger.defaultLogLevel=error

--- a/.github/workflows/reusable-build-publish.yml
+++ b/.github/workflows/reusable-build-publish.yml
@@ -76,14 +76,18 @@ jobs:
       - name: Gradle build (with version)
         id: build-and-test
         run: |
-          echo "Version specified, running clean build with no cache for ${{ inputs.version }}"
+          echo "build and test with version='${{ inputs.version }}'"
           ./gradlew clean build --no-build-cache --rerun-tasks
         env:
           RELEASE_VERSION: ${{ inputs.version }}
       - name: Look at files in verification
         if: ${{ failure() }}
         run: |
-          ls -la verification || true
+          if [ "$RUNNER_OS" == "Windows" ]; then
+            gci -Recurse verification || true
+          else
+            ls -la verification || true
+          fi
       - name: Upload hyper logs on failure
         if: ${{ failure() }}
         uses: actions/upload-artifact@v4

--- a/.gitignore
+++ b/.gitignore
@@ -27,6 +27,7 @@ pom.xml.bak
 .vscode
 .gradle
 .hyper
+.hyperd
 .kotlin
 bin
 out

--- a/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
@@ -21,7 +21,7 @@ tasks.register<de.undercouch.gradle.tasks.download.Download>("downloadHyper") {
     src(url)
     dest(project.layout.projectDirectory.file(hyperZipPath))
     overwrite(false)
-    
+
     inputs.property("hyperApiVersion", hyperApiVersion)
     inputs.property("osdetector.os", osdetector.os)
     outputs.file(project.layout.projectDirectory.file(hyperZipPath))
@@ -35,11 +35,12 @@ tasks.register<Copy>("extractHyper") {
     includeEmptyDirs = false
 
     from(zipTree(project.layout.projectDirectory.file(hyperZipPath))) {
-        include("**/lib/hyper/hyperd")
-        include("**/lib/hyper/hyperd.exe")
-        include("**/lib/**/*.dylib")
-        include("**/lib/**/*.dll")
-        include("**/lib/**/*.so")
+        when(osdetector.os) {
+            "windows" -> include("**/bin/**/*.dll", "**/bin/hyper/hyperd.exe")
+            "osx" -> include("**/lib/**/*.dylib", "**/lib/hyper/hyperd")
+            "linux" -> include("**/lib/**/*.so", "**/lib/hyper/hyperd")
+            else -> throw GradleException("Unsupported os settings. os=${osdetector.os}, arch=${osdetector.arch}, release=${osdetector.release}, classifier=${osdetector.classifier}}")
+        }
     }
 
     eachFile {
@@ -48,15 +49,30 @@ tasks.register<Copy>("extractHyper") {
 
     into(project.layout.projectDirectory.dir(hyperDir))
 
-    // Only apply Unix permissions when not on Windows
-    if (osdetector.os != "windows") {
-        filePermissions {
-            unix("rwx------")
+    filePermissions {
+        unix("rwx------")
+    }
+
+    inputs.file(project.layout.projectDirectory.file(hyperZipPath))
+
+    val hyperdDir = project.layout.projectDirectory.dir(hyperDir).asFileTree
+
+    val exe = hyperdDir.firstOrNull { it.name.contains("hyperd") }
+        ?: throw GradleException("zip missing hyperd executable")
+
+    val lib = hyperdDir.firstOrNull { it.name.contains("tableauhyperapi") }
+        ?: throw GradleException("zip missing hyperd library")
+
+    outputs.files(exe, lib)
+
+    doLast {
+        val hasExe = exe.exists()
+        val hasLib = lib.exists()
+
+        if (!hasExe || !hasLib) {
+            throw GradleException("extractHyper failed validation. hyperdMissing=$hasExe, tableauhyperapiMissing=$hasLib")
         }
     }
-    
-    inputs.file(project.layout.projectDirectory.file(hyperZipPath))
-    outputs.dir(project.layout.projectDirectory.dir(hyperDir))
 }
 
 tasks.register<Exec>("hyperd") {

--- a/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
@@ -48,13 +48,15 @@ tasks.register<Copy>("extractHyper") {
 
     into(project.layout.projectDirectory.dir(hyperDir))
 
-    filePermissions {
-        unix("rwx------")
+    // Only apply Unix permissions when not on Windows
+    if (osdetector.os != "windows") {
+        filePermissions {
+            unix("rwx------")
+        }
     }
     
     inputs.file(project.layout.projectDirectory.file(hyperZipPath))
     outputs.dir(project.layout.projectDirectory.dir(hyperDir))
-
 }
 
 tasks.register<Exec>("hyperd") {

--- a/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
@@ -5,6 +5,7 @@ plugins {
 
 val hyperApiVersion: String by project
 val hyperZipPath = ".hyper/hyper-$hyperApiVersion.zip"
+val hyperDir = ".hyperd"
 
 tasks.register<de.undercouch.gradle.tasks.download.Download>("downloadHyper") {
     group = "hyper"
@@ -20,6 +21,10 @@ tasks.register<de.undercouch.gradle.tasks.download.Download>("downloadHyper") {
     src(url)
     dest(project.layout.projectDirectory.file(hyperZipPath))
     overwrite(false)
+    
+    inputs.property("hyperApiVersion", hyperApiVersion)
+    inputs.property("osdetector.os", osdetector.os)
+    outputs.file(project.layout.projectDirectory.file(hyperZipPath))
 }
 
 tasks.register<Copy>("extractHyper") {
@@ -41,11 +46,15 @@ tasks.register<Copy>("extractHyper") {
         relativePath = RelativePath(true, name)
     }
 
-    into(project.layout.buildDirectory.dir("hyperd"))
+    into(project.layout.projectDirectory.dir(hyperDir))
 
     filePermissions {
         unix("rwx------")
     }
+    
+    inputs.file(project.layout.projectDirectory.file(hyperZipPath))
+    outputs.dir(project.layout.projectDirectory.dir(hyperDir))
+
 }
 
 tasks.register<Exec>("hyperd") {
@@ -57,9 +66,12 @@ tasks.register<Exec>("hyperd") {
         else -> "hyperd"
     }
 
-    val executable = project.layout.buildDirectory.dir("hyperd").map { it.asFile.resolve(name).absolutePath }.get()
+    val executable = project.layout.projectDirectory.dir(hyperDir).file(name).asFile.absolutePath
     val config = project.project(":jdbc-core").file("src/test/resources/hyper.yaml")
 
     commandLine(executable)
     args("--config", config.absolutePath, "run")
+
+    inputs.files(executable, config)
+    inputs.property("os", osdetector.os)
 }

--- a/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
@@ -9,65 +9,76 @@ val hyperDir = ".hyperd"
 
 tasks.register<de.undercouch.gradle.tasks.download.Download>("downloadHyper") {
     group = "hyper"
-    val urlBase = when (osdetector.os) {
-        "windows" -> "https://downloads.tableau.com/tssoftware/tableauhyperapi-cxx-windows-x86_64-release-main"
-        "osx" -> "https://downloads.tableau.com/tssoftware/tableauhyperapi-cxx-macos-arm64-release-main"
-        "linux" -> "https://downloads.tableau.com/tssoftware/tableauhyperapi-cxx-linux-x86_64-release-main"
-        else -> throw GradleException("Unsupported os settings. os=${osdetector.os}, arch=${osdetector.arch}, release=${osdetector.release}, classifier=${osdetector.classifier}}")
+
+    val os = "os=${osdetector.os}, arch=${osdetector.arch}, classifier=${osdetector.classifier}"
+
+    val osPart = when (osdetector.os) {
+        "windows" -> "windows-x86_64-release-main"
+        "linux" -> "linux-x86_64-release-main"
+        "osx" -> when (osdetector.arch) {
+            "aarch_64" -> "macos-arm64-release-main"
+            else -> "macos-x86_64-release-main"
+        }
+
+        else -> throw GradleException("Unsupported os settings. $os")
     }
 
-    val url = "$urlBase.$hyperApiVersion.zip"
+    val url = "https://downloads.tableau.com/tssoftware/tableauhyperapi-cxx-$osPart.$hyperApiVersion.zip"
+    val zip = project.layout.projectDirectory.file(hyperZipPath)
 
     src(url)
-    dest(project.layout.projectDirectory.file(hyperZipPath))
+    dest(zip)
     overwrite(false)
 
     inputs.property("hyperApiVersion", hyperApiVersion)
     inputs.property("osdetector.os", osdetector.os)
-    outputs.file(project.layout.projectDirectory.file(hyperZipPath))
+    inputs.property("osdetector.arch", osdetector.arch)
+
+    outputs.file(zip)
+
+    doLast {
+        if (!zip.asFile.exists()) {
+            throw GradleException("downloadHyper failed validation. zip=false, $os")
+        }
+    }
 }
 
 tasks.register<Copy>("extractHyper") {
     dependsOn("downloadHyper")
 
-    val os = "os=${osdetector.os}, arch=${osdetector.arch}, release=${osdetector.release}, classifier=${osdetector.classifier}"
+    val os = "os=${osdetector.os}, arch=${osdetector.arch}, classifier=${osdetector.classifier}"
 
     group = "hyper"
     duplicatesStrategy = DuplicatesStrategy.EXCLUDE
     includeEmptyDirs = false
 
     from(zipTree(project.layout.projectDirectory.file(hyperZipPath))) {
-        when(osdetector.os) {
-            "windows" -> include("**/bin/**/*.dll", "**/bin/hyper/hyperd.exe")
-            "osx" -> include("**/lib/**/*.dylib", "**/lib/hyper/hyperd")
-            "linux" -> include("**/lib/**/*.so", "**/lib/hyper/hyperd")
-            else -> throw GradleException("Unsupported os settings. $os")
-        }
+        include("**/hyperd.exe", "**/hyperd", "**/*.dll", "**/*.dylib", "**/*.so")
     }
+
+    into(project.layout.projectDirectory.dir(hyperDir))
 
     eachFile {
         relativePath = RelativePath(true, name)
     }
 
-    into(project.layout.projectDirectory.dir(hyperDir))
-
     filePermissions {
         unix("rwx------")
     }
 
-    inputs.file(project.layout.projectDirectory.file(hyperZipPath))
-
     val hyperdDir = project.layout.projectDirectory.dir(hyperDir).asFileTree
 
-    val exe = hyperdDir.firstOrNull { it.name.contains("hyperd") }
-        ?: throw GradleException("zip missing hyperd executable. $os, files=[${hyperdDir.map { it.absolutePath }}]")
-
-    val lib = hyperdDir.firstOrNull { it.name.contains("tableauhyperapi") }
-        ?: throw GradleException("zip missing hyperd library, $os, files=[${hyperdDir.map { it.absolutePath }}]")
-
-    outputs.files(exe, lib)
+    inputs.file(project.layout.projectDirectory.file(hyperZipPath))
+    outputs.files(hyperdDir.files)
 
     doLast {
+
+        val exe = hyperdDir.firstOrNull { it.name.contains("hyperd") }
+            ?: throw GradleException("zip missing hyperd executable. $os, files=${hyperdDir.map { it.absolutePath }}")
+
+        val lib = hyperdDir.firstOrNull { it.name.contains("tableauhyperapi") }
+            ?: throw GradleException("zip missing hyperd library, $os, files=${hyperdDir.map { it.absolutePath }}")
+
         if (!exe.exists() || !lib.exists()) {
             throw GradleException("extractHyper failed validation. hyperd=${exe.exists()}, lib=${lib.exists()}, $os")
         }

--- a/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/hyper-conventions.gradle.kts
@@ -30,6 +30,8 @@ tasks.register<de.undercouch.gradle.tasks.download.Download>("downloadHyper") {
 tasks.register<Copy>("extractHyper") {
     dependsOn("downloadHyper")
 
+    val os = "os=${osdetector.os}, arch=${osdetector.arch}, release=${osdetector.release}, classifier=${osdetector.classifier}"
+
     group = "hyper"
     duplicatesStrategy = DuplicatesStrategy.EXCLUDE
     includeEmptyDirs = false
@@ -39,7 +41,7 @@ tasks.register<Copy>("extractHyper") {
             "windows" -> include("**/bin/**/*.dll", "**/bin/hyper/hyperd.exe")
             "osx" -> include("**/lib/**/*.dylib", "**/lib/hyper/hyperd")
             "linux" -> include("**/lib/**/*.so", "**/lib/hyper/hyperd")
-            else -> throw GradleException("Unsupported os settings. os=${osdetector.os}, arch=${osdetector.arch}, release=${osdetector.release}, classifier=${osdetector.classifier}}")
+            else -> throw GradleException("Unsupported os settings. $os")
         }
     }
 
@@ -58,19 +60,16 @@ tasks.register<Copy>("extractHyper") {
     val hyperdDir = project.layout.projectDirectory.dir(hyperDir).asFileTree
 
     val exe = hyperdDir.firstOrNull { it.name.contains("hyperd") }
-        ?: throw GradleException("zip missing hyperd executable")
+        ?: throw GradleException("zip missing hyperd executable. $os, files=[${hyperdDir.map { it.absolutePath }}]")
 
     val lib = hyperdDir.firstOrNull { it.name.contains("tableauhyperapi") }
-        ?: throw GradleException("zip missing hyperd library")
+        ?: throw GradleException("zip missing hyperd library, $os, files=[${hyperdDir.map { it.absolutePath }}]")
 
     outputs.files(exe, lib)
 
     doLast {
-        val hasExe = exe.exists()
-        val hasLib = lib.exists()
-
-        if (!hasExe || !hasLib) {
-            throw GradleException("extractHyper failed validation. hyperdMissing=$hasExe, tableauhyperapiMissing=$hasLib")
+        if (!exe.exists() || !lib.exists()) {
+            throw GradleException("extractHyper failed validation. hyperd=${exe.exists()}, lib=${lib.exists()}, $os")
         }
     }
 }

--- a/jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/core/partial/DataCloudQueryPollingMockTests.java
+++ b/jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/core/partial/DataCloudQueryPollingMockTests.java
@@ -32,8 +32,6 @@ import lombok.val;
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.Timeout;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
 import salesforce.cdp.hyperdb.v1.HyperServiceGrpc;
 
 @Slf4j
@@ -56,20 +54,17 @@ public class DataCloudQueryPollingMockTests extends HyperGrpcTestBase {
     }
 
     @SneakyThrows
-    @ParameterizedTest
+    @Test
     @Timeout(60)
-    @ValueSource(
-            strings = {
-                "select cast(a as numeric(38,18)) a, cast(a as numeric(38,18)) b, cast(a as numeric(38,18)) c from generate_series(1, 1024 * 1024 * 1024) as s(a) order by a asc;",
-                "SELECT PG_SLEEP(10);"
-            })
-    void getQueryInfoRetriesOnTimeout(String query) {
+    @Disabled("flakey test, disabled until HyperGrpcClientExecutor interface fix")
+    void getQueryInfoRetriesOnTimeout() {
         val configWithSleep =
                 HyperServerConfig.builder().grpcRequestTimeoutSeconds("2s").build();
         try (val connection = getInterceptedClientConnection(configWithSleep)) {
             val statement = connection.createStatement().unwrap(DataCloudStatement.class);
 
-            statement.executeAsyncQuery(query);
+            statement.executeAsyncQuery(
+                    "select cast(a as numeric(38,18)) a, cast(a as numeric(38,18)) b, cast(a as numeric(38,18)) c from generate_series(1, 1024 * 1024 * 1024) as s(a) order by a asc;");
             val queryId = statement.getQueryId();
 
             verifyThat(calledMethod(HyperServiceGrpc.getGetQueryInfoMethod()), times(0));

--- a/jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/exception/QueryExceptionHandlerTest.java
+++ b/jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/exception/QueryExceptionHandlerTest.java
@@ -21,6 +21,7 @@ import static org.junit.jupiter.api.Assertions.assertInstanceOf;
 import com.salesforce.datacloud.jdbc.util.GrpcUtils;
 import io.grpc.StatusRuntimeException;
 import java.sql.SQLException;
+import lombok.val;
 import org.junit.jupiter.api.Test;
 
 class QueryExceptionHandlerTest {
@@ -32,7 +33,9 @@ class QueryExceptionHandlerTest {
 
         assertInstanceOf(SQLException.class, actualException);
         assertEquals("42P01", actualException.getSQLState());
-        assertEquals("42P01: Table not found\n" + "DETAIL:\n" + "\nHINT:\n", actualException.getMessage());
+        val sep = System.lineSeparator();
+        assertEquals(
+                "42P01: Table not found" + sep + "DETAIL:" + sep + sep + "HINT:" + sep, actualException.getMessage());
         assertEquals(StatusRuntimeException.class, actualException.getCause().getClass());
     }
 

--- a/jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/hyper/HyperServerProcess.java
+++ b/jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/hyper/HyperServerProcess.java
@@ -60,14 +60,15 @@ public class HyperServerProcess implements AutoCloseable {
     public HyperServerProcess(HyperServerConfig.HyperServerConfigBuilder config) {
         log.info("starting hyperd, this might take a few seconds");
 
-        val executable = new File("../build/hyperd/hyperd");
+        val isWindows = System.getProperty("os.name").toLowerCase().contains("win");
+        val executable = new File("../.hyperd/hyperd" + (isWindows ? ".exe" : ""));
         val yaml = Paths.get(requireNonNull(HyperTestBase.class.getResource("/hyper.yaml"))
                         .toURI())
                 .toFile();
 
         if (!executable.exists()) {
             Assertions.fail("hyperd executable couldn't be found, have you run `gradle extractHyper`? expected="
-                    + executable.getAbsolutePath());
+                    + executable.getAbsolutePath() + ", os=" + System.getProperty("os.name"));
         }
 
         val builder = new ProcessBuilder()

